utils/audioprocessing/freesound_audio_analysis.py

```diff
@@ -18,143 +18,78 @@
+
+            # Move essentia output files to analysis data directory
+            if settings.ESSENTIA_PROFILE_FILE_PATH:
+                # Never versions of FreesoundExtractor using profile file use a different naming convention
+                shutil.move(os.path.join(self.tmp_directory, 'ess_%i' % self.sound.id), statistics_path)
+                shutil.move(os.path.join(self.tmp_directory, 'ess_%i_frames' % self.sound.id), frames_path)
             else:
-                # Directory could not be created, raise exception
-                raise
-
-    def cleanup(files):
-        success("cleaning up processing files: " + ", ".join(files))
-        for filename in files:
-            try:
-                os.unlink(filename)
-            except:
-                pass
-
-    def failure(message, error=None, failure_state="FA"):
-        sound.set_analysis_state(failure_state)
-        logging_message = "ERROR: Failed to analyze sound with id %s\n" % sound.id
-        logging_message += "\tmessage: %s\n" % message
-        if error:
-            logging_message += "\terror: %s" % str(error)
-        write_log(logging_message)
-        cleanup(to_cleanup)
-
-    def success(message):
-        write_log('- ' + message)
-
-    to_cleanup = []  # This will hold a list of files to cleanup after processing
-
-    try:
-        # Get the path of the original sound
-        sound_path = sound.locations('path')
-        if settings.USE_PREVIEWS_WHEN_ORIGINAL_FILES_MISSING and not os.path.exists(sound_path):
-            sound_path = sound.locations('preview.LQ.mp3.path')
-
-        if not os.path.exists(sound_path):
-            failure('could not find file with path %s' % sound_path)
-            return False
+                shutil.move(os.path.join(self.tmp_directory, 'ess_%i_statistics.yaml' % self.sound.id), statistics_path)
+                shutil.move(os.path.join(self.tmp_directory, 'ess_%i_frames.json' % self.sound.id), frames_path)
+
+            self.log_info("created analysis files with FreesoundExtractor: %s, %s" % (statistics_path, frames_path))
+
+            # Change sound analysis and similarity states
+            self.sound.set_analysis_state('OK')
+            self.sound.set_similarity_state('PE')  # Set similarity to PE so sound will get indexed to Gaia
 
-        # Convert to mono PCM and save PCM version in `tmp_wavefile`
-        try:
-            tmp_wavefile = tempfile.mktemp(suffix=".wav", prefix=str(sound.id))
-            audioprocessing.convert_using_ffmpeg(sound_path, tmp_wavefile, mono_out=True)
-            to_cleanup.append(tmp_wavefile)
-            success("converted to PCM: " + tmp_wavefile)
         except AudioProcessingException as e:
-            failure("conversion to PCM failed", e)
-            return False
-        except IOError as e:
-            # Could not create tmp file
-            failure("could not create tmp_wavefile file", e)
+            self.failure(e)
```

Comment:
the signature of failure is (message, error). Is this parameter the message or the error (if the message, perhaps it should be `str(e)`?)